yt_dlp/extractor/jiocinema.py

```diff
@@ -0,0 +1,324 @@
+        }
+
+        payload = {
+            "appName": "RJIL_JioCinema",
+            "deviceType": "phone",
+            "os": "ios",
+            "deviceId": self.deviceId,
+            "freshLaunch": True,
+            "adId": self.deviceId,
+            "appVersion": "23.11.22.0-dc532c79"
+        }
+
+        token_response = self._download_json(url_or_request=token_endpoint, video_id=None, note='Fetching Auth Token', data=bytes(json.dumps(payload), encoding='utf8'), headers=headers)
+
+        if "authToken" in token_response:
+            return token_response["authToken"]
+
+        return None
+
+    def _get_media_metadata(self, media_id):
+        media_metadata = {}
+        metadata_endpoint_template = 'https://content-jiovoot.voot.com/psapi/voot/v1/voot-web/content/query/asset-details?&ids=include:{media_id}&responseType=common&devicePlatformType=desktop'
+
+        metadata_endpoint = metadata_endpoint_template.format(media_id=media_id)
+
+        media_metadata_response = self._download_json(metadata_endpoint, None, 'Fetching Episode Metadata')
+
+        if media_metadata_response and 'result' in media_metadata_response and media_metadata_response['result']:
+            media_metadata_response = media_metadata_response['result'][0]
+
+        media_metadata['showName'] = media_metadata_response['showName']
+        media_metadata['title'] = media_metadata_response['fullTitle']
+        media_metadata['shortTitle'] = media_metadata_response['shortTitle']
+        media_metadata['shortSynopsis'] = media_metadata_response['shortSynopsis']
+        media_metadata['fullSynopsis'] = media_metadata_response['fullSynopsis']
+        media_metadata['showName'] = media_metadata_response['showName']
+        media_metadata['season'] = media_metadata_response['season']
+        media_metadata['episode'] = media_metadata_response['episode']
+        media_metadata['multiTrackAudioEnabled'] = media_metadata_response['multiTrackAudioEnabled']
+        media_metadata['introStart'] = media_metadata_response['introStart']
+        media_metadata['introEnd'] = media_metadata_response['introEnd']
+        media_metadata['recapStart'] = media_metadata_response['recapStart']
+        media_metadata['recapEnd'] = media_metadata_response['recapEnd']
+        media_metadata['creditStart'] = media_metadata_response['creditStart']
+        media_metadata['creditEnd'] = media_metadata_response['creditEnd']
+        media_metadata['is4KSupported'] = media_metadata_response['is4KSupported']
+        media_metadata['is1080PSupported'] = media_metadata_response['is1080PSupported']
+        media_metadata['isDolbySupported'] = media_metadata_response['isDolbySupported']
+        media_metadata['hasSubtitles'] = media_metadata_response['hasSubtitles']
+        media_metadata['subtitles'] = media_metadata_response['subtitles']
```

Comment:
why can't `media_metadata_response` be returned directly? (forgot to add comment the first time)

Reply:
`media_metadata_response` has a lot of non-standard fields and there are some basic validations applied to the fields being used.

Reply:
```
from ..utils import (
    float_or_none,
    int_or_none,
    parse_age_limit,
    traverse_obj,
    unified_strdate,
) 


def _get_media_metadata(self, media_id):
        metadata_endpoint_template = 'https://content-jiovoot.voot.com/psapi/voot/v1/voot-web/content/query/asset-details?&ids=include:{media_id}&responseType=common&devicePlatformType=desktop'

        metadata_endpoint = metadata_endpoint_template.format(media_id=media_id)

        video_data = self._download_json(metadata_endpoint, None, 'Fetching Episode Metadata')

        return {
            'id': media_id,
            **traverse_obj(video_data, ('result', 0, {
                'title': ('fullTitle', {str}),
                'description': ('fullSynopsis', {str}),
                'series': ('showName', {str}),
                'season': ('seasonName', {int_or_none}),
                'season_number': ('season', {int_or_none}),
                'season_id': ('seasonId', {int_or_none}),
                'episode': ('fullTitle', {str}),
                'episode_number': ('episode', {int_or_none}),
                'timestamp': ('uploadTime', {int_or_none}),
                'release_date': ('telecastDate', {unified_strdate}),
                'release_year': ('releaseYear', {unified_strdate}),
                'age_limit': ('ageNemonic', {parse_age_limit}),
                'parentalRating': ('ageNumeric', {int_or_none}),
                'duration': ('duration', {float_or_none}),
                'duration_formated': ('durationFormatted', {float_or_none}),
                'genre': ('genres'),
                'languages': ('languages')
            })),
        }
```
        
this will be more easy for other people

Reply:
```

from ..utils import (
    float_or_none,
    int_or_none,
    parse_age_limit,
    traverse_obj,
    unified_strdate,
)

def _get_media_metadata(self, media_id):
        metadata_endpoint_template = 'https://content-jiovoot.voot.com/psapi/voot/v1/voot-web/content/query/asset-details?&ids=include:{media_id}&responseType=common&devicePlatformType=desktop'

        metadata_endpoint = metadata_endpoint_template.format(media_id=media_id)

        video_data = self._download_json(metadata_endpoint, None, 'Fetching Episode Metadata')

        return {
            'id': media_id,
            **traverse_obj(video_data, ('result', 0, {
                'title': ('fullTitle', {str}),
                'description': ('fullSynopsis', {str}),
                'series': ('showName', {str}),
                'season': ('seasonName', {int_or_none}),
                'season_number': ('season', {int_or_none}),
                'season_id': ('seasonId', {int_or_none}),
                'episode': ('fullTitle', {str}),
                'episode_number': ('episode', {int_or_none}),
                'timestamp': ('uploadTime', {int_or_none}),
                'release_date': ('telecastDate', {unified_strdate}),
                'release_year': ('releaseYear', {unified_strdate}),
                'age_limit': ('ageNemonic', {parse_age_limit}),
                'parentalRating': ('ageNumeric', {int_or_none}),
                'duration': ('duration', {float_or_none}),
                'duration_formated': ('durationFormatted', {float_or_none}),
                'genre': ('genres'),
                'languages': ('languages')
            })),
        }
        
```


this format of metadata will be easy for others to understand the code

Reply:
```

from ..utils import (
    float_or_none,
    int_or_none,
    parse_age_limit,
    traverse_obj,
    unified_strdate,
)

def _get_media_metadata(self, media_id):
        metadata_endpoint_template = 'https://content-jiovoot.voot.com/psapi/voot/v1/voot-web/content/query/asset-details?&ids=include:{media_id}&responseType=common&devicePlatformType=desktop'

        metadata_endpoint = metadata_endpoint_template.format(media_id=media_id)

        video_data = self._download_json(metadata_endpoint, None, 'Fetching Episode Metadata')

        return {
            'id': media_id,
            **traverse_obj(video_data, ('result', 0, {
                'title': ('fullTitle', {str}),
                'description': ('fullSynopsis', {str}),
                'series': ('showName', {str}),
                'season': ('seasonName', {int_or_none}),
                'season_number': ('season', {int_or_none}),
                'season_id': ('seasonId', {int_or_none}),
                'episode': ('fullTitle', {str}),
                'episode_number': ('episode', {int_or_none}),
                'timestamp': ('uploadTime', {int_or_none}),
                'release_date': ('telecastDate', {unified_strdate}),
                'release_year': ('releaseYear', {unified_strdate}),
                'age_limit': ('ageNemonic', {parse_age_limit}),
                'parentalRating': ('ageNumeric', {int_or_none}),
                'duration': ('duration', {float_or_none}),
                'duration_formated': ('durationFormatted', {float_or_none}),
                'genre': ('genres'),
                'languages': ('languages')
            })),
        }
        
```


this format of metadata will be easy for others to understand the code